main.nf

```diff
@@ -1,234 +1,292 @@
+        |
+        \ -> "Undeterminded/<Project>" -> FastQC        -> MultiQCPerFlowcell + MultiQCPerProject
+                                       -> FastqScreen   -> MultiQCPerFlowcell + MultiQCPerProject
+    */
+
+    get:
+        run_folder
+
+    main:
+        interop_summary(run_folder)
+        get_QC_thresholds(run_folder)
+        get_metadata(run_folder)
+        project_and_reads = get_project_and_reads(params.run_folder)
+        fastqc(project_and_reads)
+        fastq_screen(project_and_reads)
+        multiqc_per_flowcell( params.run_folder,
+            fastqc.out.map{ it[1] }.collect().ifEmpty([]),
+            fastq_screen.out.map{ it[1] }.collect().ifEmpty([]),
+            interop_summary.out.map{ it[1] }.collect().ifEmpty([]),
+            get_QC_thresholds.out.collect(),
+            get_metadata.out.collect())
+        multiqc_per_project( params.run_folder,
+            combine_results_by_project(fastqc.out.groupTuple(),fastq_screen.out.groupTuple()),
+            get_metadata.out.collect())
+
+    emit:
+        flowcellqc = multiqc_per_flowcell.out
+        projectqc = multiqc_per_project.out
+
+}
 
 // ---------------------------------------------------
 // Processes
 // ---------------------------------------------------
-process InteropSummary {
+
+process fastqc {
 
     input:
-    file runfolder
+    tuple project, path(fastq_file)
 
     output:
-    file runfolder_summary_interop into interop_summary_results
+    tuple project, path("*_fastqc.{zip,html}")
 
+    script:
     """
-    summary --csv=1 $runfolder > runfolder_summary_interop
+    fastqc -t ${task.cpus} $fastq_file
     """
 }
 
-process Fastqc {
+process fastq_screen {
 
     input:
-    set val(project), file(fastq_file) from input_fastqc
+    tuple project, path(fastq_file)
 
     output:
-    set val(project), file("*_fastqc.{zip,html}") into fastqc_results
+    tuple project, path("*_screen.{txt,html}")
 
+    script:
     """
-    fastqc $fastq_file
+    sed -E 's/^(THREADS[[:blank:]]+)[[:digit:]]+/\1${task.cpus}/' \\
+        ${params.config_dir}/fastq_screen.conf > fastq_screen.conf
+    if [ ! -e "${params.fastqscreen_databases}" ]; then
+        fastq_screen --get_genomes
+    elif [ "${params.fastqscreen_databases}" != "${fastqscreen_default_databases}" ]; then
+        sed -i 's#${fastqscreen_default_databases}#${params.fastqscreen_databases}#' fastq_screen.conf
+    fi
+    fastq_screen --conf fastq_screen.conf $fastq_file
     """
 }
 
-fastqc_results.into{ fastqc_results_for_flowcell;  fastqc_results_for_project_ungrouped }
-
-process FastqScreen {
+process get_QC_thresholds {
 
     input:
-    set val(project), file(fastq_file) from input_fastqscreen
-    file config_dir from config_dir
-    file db from fastq_screen_db
+    path runfolder
 
     output:
-    set val(project), file("*_screen.{txt,html}") into fastq_screen_results
+    path "qc_thresholds.yaml"
```

Comment:
This output should be optional, otherwise the flowcell report won't be generated. I solved it like this in current master (It's not shown here due to the merge conflict) :   `file("qc_thresholds.yaml") optional true into qc_thresholds_result`

Reply:
It's the same syntax more or less. Added `optional true`.